renku/cli/dataset.py

```diff
@@ -490,6 +522,84 @@ def remove(client, names):
     click.secho('OK', fg='green')
 
 
+def tag_dataset(client, name, tag, description, force=False):
```

Comment:
To fit the new structure better, you can decorate this function with `pass_local_client`, since its a library part. That way click commands are only decorated with click stuff and we keep library contained.